### ISSUE TITLE
No color, only gray.

### ISSUE THREAD

**Issue by cbmeeks** [InPermutation/tms9918a-arduino#1]
Hi there.

I've implemented pretty much this exact design.  Just trying to get the colored backgrounds.

The only difference in what I've implemented and what you have on this video is that I am missing the ferrite bead.  But I can't imagine that would cause huge issues.

I can see the VPD cycle through the colors...but instead of actually being color, they are all various shades of gray.  On my oscilloscope,  I can see what looks to be a proper NTSC pattern including color burst.  But then with some backgrounds the signal goes all wonky.

I've tried four different TMS9918's and they all have the same issue.

Any ideas what might be wrong?  Not sure where else to ask.  :-)

Thanks.

**Comment by InPermutation** [comment]
It may be your TV is too sensitive to timing differences between the NTSC spec and the actual output of the chip – can you try it on a different TV?

**Comment by cbmeeks** [comment]
Thanks for the reply.

I did try on another TV.  Same results.  

In fact, on this same TV, I connected a TI-99/4a and it worked great.   I'm really wondering if my timing is off.

I tested the signal on my oscilloscope and it looked OK.  Although, many times it would get all garbled up.  But when the signal was clean, the timings looked good.  I checked the colorbust timing as well and it looked almost perfect.

I also tried five different TMS9918 chips.  Including one that I pulled from a working TI-99/4a.  All show the same results.

So this really has me baffled.

I've ordered the RC2014 board that you have and I've ordered another round of crystals that jblang suggests on his GitHub site.  Although, the crystals I have came from Mouser and Digikey so they should be accurate.

**Comment by InPermutation** [comment]
What do you have in place of the ferrite bead? It's the second-to-last step in the video circuit, but I don't have the electronics knowledge to understand exactly what the LC circuit is doing. Maybe it's important.

Once you get the RC2014 board, there should be a bead included – try using it and seeing if that fixes the missing color?

**Comment by cbmeeks** [comment]
I don't have a ferrite bead yet.  I have one on order.
I didn't order the board from Tindie.  I just sent the gerbers to JLCPCB.  They have already been shipped.  Other than the ferrite bead and the PCB, I have everything already.

Just not sure what I'm doing wrong.

**Comment by InPermutation** [comment]
In your circuit, what is connected to the collector of Q1?

<img width="159" alt="image" src="https://user-images.githubusercontent.com/1096993/160005165-ed927d91-1ab2-451b-95a5-2cb9dd2d80d8.png">

**Comment by cbmeeks** [comment]
It has a wire to VCC that has a 0.1uF cap next to it.

As far as I can tell, my wiring is 99% exact to that pic with the exception of the ferrite bead.

**Comment by cbmeeks** [comment]
FYI, I found some ferrite beads.  Same deal.

I put the crystal on my logic analyzer and, weirdly, it's reporting the frequency at 10MHz and 12.5MHz.  Then, it was 10 and 11.11111MHz.  

I might actually solder up a small board just to see if it's the breadboard.  But I don't see how.  Leads are very short and yours works.

Ugh...

![image](https://user-images.githubusercontent.com/49341/160022983-5a17c91a-5f7c-4027-81cb-496a893f6d31.png)

![image](https://user-images.githubusercontent.com/49341/160023044-94692790-473c-4b6e-8b2b-668b222ee92b.png)

**Comment by InPermutation** [comment]
1. What kind of breadboard are you using? I have heard many [horror stories](https://wilsonminesco.com/6502primer/construction.html) of people building stuff with >2MHz signals that then fails because of the breadboard connections. It has worked so far for me. I'm using the expensive BB830s that come with Ben Eater's kits. But, I could see the breadboard affecting the capacitance somehow.
2. The capacitors used on the crystal oscillator circuit are ridiculously tiny (16pF !) - the stray capacitance added by your logic analyzer's probes may be causing the frequency to change. Try measuring one of the VDP's clock output pins: `CPUCLK` and `GROMCLK` (should be 3.58MHz and 447 kHz, respectively).
3. The datasheet says you should be able to tie COMVID directly to the TV with a 330Ω pulldown:
<img width="616" alt="image" src="https://user-images.githubusercontent.com/1096993/160188938-80954318-9b79-4ed2-9255-4db59496aa80.png">
I've never tried this, but it may help with your gray video problem.

**Comment by cbmeeks** [comment]
I'm using the same BB830's.  In fact, the breadboard is brand new and has only had this circuit.

I've tried the same circuit from the TI docs as well (same results).

I haven't done the exact math, but it seems the GROMCLK and CLK are correct (well, they have the correct ratio).

I'm getting another crystal next week.  It's the only one from Mouser with that frequency that I don't already have.  So maybe that will fix it.

I also soldered the circuit to a proto-board and got the same results as well.

**Comment by cbmeeks** [comment]
I finally got it to work!

Turns out, the 10.73866MHz crystals I got before (one from Mouser and one from Digikey) were not accurate enough.

As a last-ditch effort, I found another version on Mouser that is 

    10.73866MHz 
    16pF -20

And I now have color!

For those that run into this issue, here is the Mouser part number I used:

    449-LFXTAL029962REEL 

Manufacturer part number:
    
    LFXTAL029962Reel

**Comment by cbmeeks** [comment]
![image](https://user-images.githubusercontent.com/49341/160465750-ab981500-3e09-46a9-8d04-b656bbcff2b9.png)

![image](https://user-images.githubusercontent.com/49341/160465833-736cac96-00d6-4df2-a022-cf576756068b.png)

![image](https://user-images.githubusercontent.com/49341/160465916-49c58cfd-51c5-4515-a0d5-b279d3853eb7.png)

**Comment by InPermutation** [comment]
Congratulations! That is a beautiful purple.

**Comment by OldScot** [comment]
I had the same problem (no colour) on the composite output. I was trying to revive a "Mikrokolor 80" board I had built in the 80s.
That circuit uses a 33pf and a 5-60pf cap on the crystal legs. No adjustment on the 5-60 would work.
First I tried replacing the 5-60 cap (which off the board tested good...) with no effect.
Then I tried replacing the existing crystal. Adjusted the 5-60 cap and ta-da, colour!
The Mouser part was out of stock.
I ended up ordering the crystal from Arcade Parts and Repair. Super fast turnaround and delivery! Recommended.
[10.738635 Crystal](https://www.arcadepartsandrepair.com/store/pinball-kits-parts/pinball-parts/10-738635mhz-crystal-qc1009/)
